src/Board.js

```diff
@@ -1,52 +1,169 @@
+        }
+        } else {
+            handleCellClick(rowIndex, colIndex);
+        }
+        setStartCell(null);
+    };
+
+    const handleDragSelect = (rowIndex, colIndex) => {
+        const updatedGrid = [...grid];
+        const cell = updatedGrid[rowIndex][colIndex];
+
+        if (!cell.isSelected) {
+            setChosenLetters(prevLetters => prevLetters + cell.letter);
+            setSelectedCells(prevCells => [...prevCells, { row: rowIndex, col: colIndex }]);
+            cell.isSelected = true;
+            setGrid(updatedGrid);
+        }
+    };
+
+    const handleCellClick = (rowIndex, colIndex) => {
+        if (!isDragging) {
+            const updatedGrid = [...grid];
+            const cell = updatedGrid[rowIndex][colIndex];
+
+            if (selectedCells.length === 0) {
+                setChosenLetters('');
+            }
+            
+            if (cell.isSelected) {
+                checkWord();
+            } else {
+                setChosenLetters(prevLetters => prevLetters + cell.letter);
+                setSelectedCells(prevCells => [...prevCells, { row: rowIndex, col: colIndex }]);
+                cell.isSelected = !cell.isSelected;
+            }
+
+            setGrid(updatedGrid);
+        }
+    };
+
+    const checkWord = () => {
+        const updatedGrid = [...grid];
+        if (words.includes(chosenLetters)) {
+            setChosenLetters('valid word');
+            selectedCells.forEach(({ row, col }) => {
+                updatedGrid[row][col].isFound = true;
+                updatedGrid[row][col].isThemeWord = true;
+                updatedGrid[row][col].isSelected = false;
+            });
+        } else if (spangram === chosenLetters) {
+            selectedCells.forEach(({ row, col }) => {
+                updatedGrid[row][col].isFound = true;
+                updatedGrid[row][col].isSpangram = true;
+                updatedGrid[row][col].isSelected = false;
+            });
+        } else {
+            setChosenLetters('Not in word list');
+            selectedCells.forEach(({ row, col }) => {
+                updatedGrid[row][col].isSelected = false;
+            });
+        }
+        setSelectedCells([]);
+        setGrid(updatedGrid);
+    };
 
     return (
         <div className="grid">
+            <h1> {chosenLetters} </h1>
             {grid.map((row, rowIndex) => (
                 <div key={rowIndex} className="row">
                     {row.map((cell, colIndex) => (
-                        <Cell key={colIndex} cell={cell} onclick="chooseLetter()"/>
+                        <Cell key={colIndex}
+                        cell={cell}
+                        onClick={() => handleCellClick(rowIndex, colIndex)}
+                        onMouseDown={() => handleMouseDown(cell, rowIndex, colIndex)}
+                        onMouseEnter={() => handleMouseEnter(rowIndex, colIndex)}
+                        onMouseUp={() => handleMouseUp(rowIndex, colIndex)}/>
```

Comment:
I see what's happening here. Usually I wouldn't mind the logic living here. Logic being handleMouseDown, handleMouseEnter, etc. Buttttt, it shouldn't live here. That logic belongs in the `<Cell />` component

```suggestion
                        cell={cell}
                        position={[rowIndex, colIndex]}
                        setIsDragging={setIsDragging}
                        setChosenLetters={setChosenLetters}
                        setSelectedCells={setSelectedCells}
                        ...
```

There is argument to be made that is more props, but doing that allows the logic `handleMouseDown` etc. to live in the `<Cell />` component overall reducing the bloat on this `<Board />` component.

Remember this board component is the most high level component. Functional Logic that should live in a children should live in the children. The high level component should contain high level logic like the state and `checkWord` can actually live in this component still